modules/dynamic-operator/api/v1alpha1/kuberlogicservice_types.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	v11 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const readyCondType = "Ready"
+
+// KuberLogicServiceStatus defines the observed state of KuberLogicService
+type KuberLogicServiceStatus struct {
+	Conditions []metav1.Condition `json:"conditions"`
+}
+
+type KuberLogicServiceSpec struct {
+	// Type of the cluster
+	// +kubebuilder:validation:Enum=postgresql;mysql;redis
+	Type string `json:"type"`
+	// Amount of replicas
+	// +kubebuilder:validation:Maximum=5
+	Replicas int32 `json:"replicas"`
+	// Volume size
+	VolumeSize string `json:"volumeSize,omitempty"`
+	// 2 or 3 digits: 5 or 5.7 or 5.7.31
+	// +kubebuilder:validation:Pattern=^\d+[\.\d+]*$
+	Version string `json:"version,omitempty"`
+
+	// any advanced configuration is supported
+	Advanced v11.JSON `json:"advanced,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:printcolumn:name="Ready",type="string",JSONPath=".status.conditions[?(@.type == 'Ready')].status",description="The cluster readiness"
+// +kubebuilder:printcolumn:name="Status",type="string",JSONPath=".status.conditions[?(@.type == 'Ready')].reason",description="The cluster status"
+// +kubebuilder:printcolumn:name="Type",type=string,JSONPath=`.spec.type`,description="The cluster type"
+// +kubebuilder:printcolumn:name="Replicas",type=integer,JSONPath=`.spec.replicas`,description="The number of desired replicas"
+// +kubebuilder:printcolumn:name="Volume",type=string,JSONPath=`.spec.volumeSize`,description="Volume size for the cluster"
+// +kubebuilder:printcolumn:name="CPU Request",type=string,JSONPath=`.spec.resources.requests.cpu`,description="CPU request"
+// +kubebuilder:printcolumn:name="Memory Request",type=string,JSONPath=`.spec.resources.requests.memory`,description="Memory request"
```

Comment:
- won't work unless we move requests out of advanced conf
- yes, it could be received from advanced, but still it does not specified in advanced conf